cpp/daal/include/services/daal_defines.h

```diff
@@ -50,7 +50,8 @@
 #endif
 
 /* Intel(R) oneDAL 64-bit integer types */
-#if (!defined(__INTEL_COMPILER)) & defined(_MSC_VER)
+
+#if (!defined(__INTEL_COMPILER) || !defined(__INTEL_LLVM_COMPILER)) && defined(_MSC_VER)
```

Comment:
Could you spell out the condition for me? This truth table looks unexpected.
![image](https://user-images.githubusercontent.com/9201869/234787577-0ebb18cd-3614-4746-b1d0-17e8fcb4d7ca.png)

Are we sure we're not checking for

```c
!(defined(__INTEL_COMPILER) || defined(__INTEL_LLVM_COMPILER))
```